pages/home.tsx

```diff
@@ -1,11 +1,98 @@
 import Head from 'next/head'
-
+import theme from '../styles/theme'
+import useSWR from 'swr'
 import { withServerSideAuth } from '@clerk/nextjs/ssr'
 import { SSRUser } from '../helpers/user-details'
 import Page from '../components/Page'
-import { ServerPageProps } from '../types/types'
-import { Box, Text } from 'grommet'
+import { ServerPageProps, Status } from '../types/types'
+import {
+  Box,
+  Text,
+  Tabs,
+  Tab,
+  Card,
+  CardHeader,
+  CardBody,
+  CardFooter,
+  Button,
+  Layer
+} from 'grommet'
+import { useEffect, useState } from 'react'
+import { PrescriptionAndLocation, PharmacyLocation } from '../types/types'
+import QRCodeModal from '../components/QrCode'
+import PatientPrescriptions from '../components/PatientPrescriptions'
+import PharmacyLocationView from '../components/PharmacyLocation'
+import { Prescription } from '.prisma/client'
+import Skeleton from 'react-loading-skeleton'
+
 const Home = ({ user }: ServerPageProps) => {
+  const [prescriptions, setPrescriptions] = useState<
+    Array<PrescriptionAndLocation>
+  >([])
+  const [previousPrescriptions, setPreviousPrescriptions] = useState<
+    Array<PrescriptionAndLocation>
+  >([])
+  const [pharmacyLocation, setPharmacyLocation] = useState<PharmacyLocation>()
+  const [location, setLocation] = useState<Location>()
+  const [loading, setLoading] = useState(true)
+  const [qrCode, setQrCode] = useState('')
+  const [showQrCode, setShowQrCode] = useState(false)
+
+  const fetcher = (userRole: string) => {
+    if (userRole === 'patient') {
+      return fetch('/api/prescriptions/read', {
+        method: 'POST',
+        body: JSON.stringify({
+          patientId: user.Patient?.id
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          let prevPrescriptions: PrescriptionAndLocation[] = []
+          let activePrescriptions: PrescriptionAndLocation[] = []
+
+          data.prescription?.forEach(
+            (prescription: PrescriptionAndLocation) => {
+              if (prescription.status === Status.AwaitingPickup) {
+                activePrescriptions.push(prescription)
+              } else if (prescription.status === Status.PickupCompleted) {
+                prevPrescriptions.push(prescription)
+              }
+            }
+          )
+          setPreviousPrescriptions(prevPrescriptions)
+          setPrescriptions(activePrescriptions)
+        })
+        .catch()
+        .finally(() => setLoading(false))
+    } else if (userRole === 'staff') {
+      return fetch('/api/locations/read', {
+        method: 'POST',
+        body: JSON.stringify({
+          id: user.Staff?.locationId
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          setPharmacyLocation(data.location)
+        })
+        .catch()
+        .finally(() => setLoading(false))
+    }
+  }
+
+  const { data, error } = useSWR(user.role, fetcher)
```

Comment:
You might want to put this in a hook and call it `usePrescriptions` and `useLocation` and then use it in the patient prescriptions component. It will make this a lot easier to read I think you can have it return some nice things imagine using it like
just create and call this hook in `components/PatientPrescriptions.tsx` and then all you need to pass in is the `user`
```tsx
{user.role ===Role.Patient && <PatientPrescriptions user={user}>}
```
```tsx
const { awaitingPickup, pickupCompleted, isLoading, isError } = usePrescriptions(user)
```

then in `components/PharmacyLocation.tsx` call a hook that looks like this:
```tsx
const { location, isLoading, isError } = useLocation(user)
```
```tsx
{user.role ===Role.Staff && <PharmacyLocationView user={user}>}
```

this change will make sure that we are only ever creating stateful elements that intend to use, reduces the number of conditional renders, and mean that you don't have to store any sort of state because we are getting it from the hooks. LMK if this makes sense. I am not gonna block the PR from merging, so up to you if you want to make this change before merging it.